src/cpp/rtps/transport/shared_mem/MultiProducerConsumerRingBuffer.hpp

```diff
@@ -322,6 +352,27 @@ class MultiProducerConsumerRingBuffer
         return (loop_flag << 31) | value;
     }
 
+        uint32_t pointer_to_head(
+            const Pointer& pointer) const
+    {
+        // Init the head as write pointer in previous loop
+        uint32_t head = pointer.write_p ^ 0x80000000;
+
+        uint32_t value = head & 0x7FFFFFFF;
+        uint32_t loop_flag = head >> 31;
+
+        if (value +  pointer.free_cells >= node_->total_cells_)
+        {
+            loop_flag ^= 1;
+        }
+
+        // Skip the free cells
+        value = value + pointer.free_cells % node_->total_cells_;
```

Comment:
I prefer to explicitly state operator order:
```suggestion
        value = value + (pointer.free_cells % node_->total_cells_);
```

Reply:
better:
(value + pointer.free_cells) % node_->total_cells_;

Reply:
That's why I always put the parenthesis